src/main/java/org/opensearch/knn/index/query/iterators/ByteVectorIdsKNNIterator.java

```diff
@@ -17,30 +17,32 @@
  * Inspired by DiversifyingChildrenFloatKnnVectorQuery in lucene
  * https://github.com/apache/lucene/blob/7b8aece125aabff2823626d5b939abf4747f63a7/lucene/join/src/java/org/apache/lucene/search/join/DiversifyingChildrenFloatKnnVectorQuery.java#L162
  *
- * The class is used in KNNWeight to score filtered KNN field by iterating filterIdsArray.
+ * The class is used in KNNWeight to score all docs, but, it iterates over filterIdsArray if filter is provided
  */
-public class FilteredIdsKNNByteIterator implements KNNIterator {
-    // Array of doc ids to iterate
-    protected final BitSet filterIdsBitSet;
+public class ByteVectorIdsKNNIterator implements KNNIterator {
     protected final BitSetIterator bitSetIterator;
     protected final byte[] queryVector;
     protected final KNNBinaryVectorValues binaryVectorValues;
     protected final SpaceType spaceType;
     protected float currentScore = Float.NEGATIVE_INFINITY;
     protected int docId;
 
-    public FilteredIdsKNNByteIterator(
+    public ByteVectorIdsKNNIterator(
         final BitSet filterIdsBitSet,
```

Comment:
should we make filterIdsBitSet as Optional? or atleast annotate as @Nullable to provide a signal that this parameter can be null.

Reply:
Ack